src/Microsoft.AspNetCore.Server.Testing/Deployers/NanoDeployer/NanoDeployer.cs

```diff
@@ -0,0 +1,251 @@
+            }
+            catch { }
+
+            try
+            {
+                Logger.LogInformation($"Deleting the deployed folder '{_deployedFolderPathInFileShare}'");
+                Directory.Delete(_deployedFolderPathInFileShare, recursive: true);
+            }
+            catch { }
+
+            try
+            {
+                Logger.LogInformation($"Deleting the locally published folder '{DeploymentParameters.PublishedApplicationRootPath}'");
+                Directory.Delete(DeploymentParameters.PublishedApplicationRootPath, recursive: true);
+            }
+            catch { }
+        }
+
+        private void RunScript(string serverAction)
+        {
+            var parameterBuilder = new StringBuilder();
+            parameterBuilder.Append($"\"{_remotePSSessionHelperScript}\"");
+            parameterBuilder.Append($" -serverName {_deploymentParameters.ServerName}");
+            parameterBuilder.Append($" -accountName {_deploymentParameters.ServerAccountName}");
+            parameterBuilder.Append($" -accountPassword {_deploymentParameters.ServerAccountPassword}");
+            parameterBuilder.Append($" -executablePath {Path.Combine(_deployedFolderPathInFileShare, _deploymentParameters.NanoServerRelativeExecutablePath)}");
+            parameterBuilder.Append($" -serverType {_deploymentParameters.ServerType}");
+            parameterBuilder.Append($" -serverAction {serverAction}");
+
+            // todo: launch a powershell process to make the website point to the created folder
+            var startInfo = new ProcessStartInfo
+            {
+                FileName = "powershell.exe",
+                Arguments = parameterBuilder.ToString(),
+                UseShellExecute = false,
+                CreateNoWindow = true,
+                RedirectStandardError = true,
+                RedirectStandardOutput = true,
+                RedirectStandardInput = true
+            };
+
+            using (var deployToNanoProcess = new Process() { StartInfo = startInfo })
+            {
+                deployToNanoProcess.EnableRaisingEvents = true;
+                deployToNanoProcess.ErrorDataReceived += (sender, dataArgs) =>
+                {
+                    if (!string.IsNullOrEmpty(dataArgs.Data))
+                    {
+                        Logger.LogWarning("Nano server: " + dataArgs.Data);
+                    }
+                };
+
+                deployToNanoProcess.OutputDataReceived += (sender, dataArgs) =>
+                {
+                    if (!string.IsNullOrEmpty(dataArgs.Data))
+                    {
+                        Logger.LogInformation("Nano server: " + dataArgs.Data);
+                    }
+                };
+
+                deployToNanoProcess.Start();
+                deployToNanoProcess.BeginErrorReadLine();
+                deployToNanoProcess.BeginOutputReadLine();
+                deployToNanoProcess.WaitForExit((int)TimeSpan.FromMinutes(1).TotalMilliseconds);
```

Comment:
Let me see if I understand the logic here... So, you invoke the script, wait for it for 1 minute and then kill it? Will killing the process/the powershell session kill the remote process (the server) too?